src/quest_manager/views.py

```diff
@@ -1240,12 +1244,17 @@ def ajax_save_draft(request):
         # xp_requested = request.POST.get('xp_requested')
 
         sub = get_object_or_404(QuestSubmission, pk=submission_id)
+        # if there is no draft comment, then the quest is not in progress
+        if not sub.draft_comment:
+            raise Http404("No draft comment found. The quest is not in progress.")
```

Comment:
The existence or non existence of a draft comment is not how we should be checking the status of a submission. We should use a method or property for that purpose. The same way other views check.

Reply:
Also, how would this happen? This line isn't tested.

Reply:
I added this as an additional check to ensure that at this point, they actually have created a comment. I'll admit that I wasn't thinking about how it could occur. I'm not sure if it's possible to make a submission without having a draft_comment with the current implementation. Maybe if they tried submitting before our update using a headless web browser? But should I just remove this?

I could definitely write a test that would trigger this by creating a submission without a draft_comment and then trying to save it as a draft.

Reply:
```py
def test_complete_no_draft_comment(self):
        """ When a student tries to complete a quest, their submission should have a draft_comment
        attribute. If it doesn't the submission is invalid, so they are shown a 404 page.
        """
        invalid_sub = baker.make(QuestSubmission, user=self.test_student, quest=self.quest,
                                 draft_comment=None)
        response = self.client.post(
            reverse('quests:complete', args=[invalid_sub.id]),
            data={'comment_text': "test comment", 'complete': True}
        )
        self.assertEqual(response.status_code, 404)
```